### PR TITLE
Check for `GLX_EXT_swap_control`

### LINKED ISSUE #967: `SwapIntervalEXT` causes SIGSEV on mesa
Using `glutin 0.9`, I get a SIGSEV only if

* I request vsync (because otherwise `SwapIntervalEXT` isn't called
* I'm using the nightly toolchain of Rust

`SwapIntervalSGI` does not crash with my driver (`Mesa 17.0.7`).

(I'm creating this issue after I realized https://github.com/amethyst/amethyst/issues/512 isn't only a problem in Amethyst, but in glutin)

Given that it does only crash on nightly, I'm not sure if this is a driver problem or a bug in Rust.

---

```
GPU: Intel® HD Graphics 620 (Kabylake GT2) 
Driver: Mesa 17.0.7
Rustc version: 1.24.0-nightly Nightly
Rustc git commit: 7eb64b86ce44cc1828dd176a8b981e37ea08fc38
```

### PR DESCRIPTION
**Pull request by torkleyy**
This is necessary because `SwapIntervalEXT` is reported as loaded even though it's unsupported.

Fixes #967

### CODE REVIEW

File: src/api/glx/mod.rs [hunk 1]
```diff
@@ -246,7 +246,8 @@ impl<'a> ContextPrototype<'a> {
         if self.opengl.vsync {
             unsafe { self.glx.MakeCurrent(self.display as *mut _, window, context) };
 
-            if extra_functions.SwapIntervalEXT.is_loaded() {
+            let ext_sc = self.extensions.split(' ').find(|&i| i == "GLX_EXT_swap_control").is_some();
+            if ext_sc && extra_functions.SwapIntervalEXT.is_loaded() {
```
Comment (torkleyy) [inline src/api/glx/mod.rs]: Not sure if I should do the same check for the `SGI` extension.

Comment (tomaka) [reply src/api/glx/mod.rs]: Yes please!

**Comment by torkleyy** [comment]
@tomaka Done.

**Comment by tomaka** [comment]
Thanks!

**Pull request merged**